plugins/woocommerce/src/Admin/Features/BlockEditorFeatureEnabled.php

```diff
@@ -29,6 +30,37 @@ public function __construct() {
 			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_styles' ) );
 			add_action( 'get_edit_post_link', array( $this, 'update_edit_product_link' ), 10, 2 );
 		}
+		if ( Features::is_enabled( self::FEATURE_ID ) ) {
+			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+		}
+	}
+
+	/**
+	 * Enqueue scripts needed for the product form block editor.
+	 */
+	public function enqueue_scripts() {
+		if ( ! PageController::is_admin_or_embed_page() ) {
+			return;
+		}
+		$post_type_object     = get_post_type_object( 'product' );
+		$block_editor_context = new WP_Block_Editor_Context( array( 'name' => 'core/edit-post' ) );
+
+		$editor_settings = array();
+		if ( ! empty( $post_type_object->template ) ) {
+			$editor_settings['template']     = $post_type_object->template;
+			$editor_settings['templateLock'] = ! empty( $post_type_object->template_lock ) ? $post_type_object->template_lock : false;
+		}
+
+		$editor_settings = get_block_editor_settings( $editor_settings, $block_editor_context );
+
+		$script_handle = 'wc-admin-edit-product';
+		wp_register_script( $script_handle, '', array(), '0.1.0', true );
+		wp_enqueue_script( $script_handle );
+		wp_add_inline_script(
+			$script_handle,
+			'var productBlockEditorSettings = productBlockEditorSettings || ' . wp_json_encode( $editor_settings ) . ';',
```

Comment:
We discussed briefly today in our call, but we might update this to `blockProductEditorSettings` and probably the same for this file name to avoid any ambiguity in naming.

We can discuss over Slack though; no reason to hold up this PR.